File: datalib/src/main/java/nl/acidcats/tumblrlikes/core/usecases/appsetup/AppSetupUseCaseImpl.java
package nl.acidcats.tumblrlikes.core.usecases.appsetup;

import android.support.annotation.NonNull;
import android.text.TextUtils;

import javax.inject.Inject;

import nl.acidcats.tumblrlikes.core.repositories.AppDataRepository;
import rx.Observable;

/**
 * Created on 17/10/2018.
 */
public class AppSetupUseCaseImpl implements AppSetupUseCase {

    private AppDataRepository _appDataRepository;

    @Inject
    AppSetupUseCaseImpl(AppDataRepository appDataRepository) {
        _appDataRepository = appDataRepository;
    }

    @NonNull
    @Override
    public Observable<Boolean> isSetupComplete() {
        String apiKey = _appDataRepository.getTumblrApiKey();
        String blog = _appDataRepository.getTumblrBlog();

        return Observable.just(!TextUtils.isEmpty(apiKey) && !TextUtils.isEmpty(blog));
    }
}

File: datalib/src/main/kotlin/nl/acidcats/tumblrlikes/core/usecases/appsetup/AppSetupUseCaseImpl.kt
package nl.acidcats.tumblrlikes.core.usecases.appsetup

import nl.acidcats.tumblrlikes.core.repositories.AppDataRepository
import rx.Observable
import javax.inject.Inject

/**
 * Created on 24/10/2018.
 */
class AppSetupUseCaseImpl @Inject constructor(private val appDataRepository: AppDataRepository) : AppSetupUseCase {

    override fun isSetupComplete(): Observable<Boolean> {
        val apiKey = appDataRepository.getTumblrApiKey()
        val blog = appDataRepository.getTumblrBlog()

        return Observable.just(!apiKey.isEmpty() && !blog.isEmpty());
    }
}
